core/lib/rom/plugins.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'dry/container'
+require 'rom/plugin'
+
+module ROM
+  # Registry of all known plugin types (command, relation, mapper, etc)
+  #
+  # @api private
+  module Plugins
+    @registry = ::Dry::Container.new
```

Comment:
why not just `extend Dry::Container::Mixin` and call it a day?

Reply:
no objection against it, I just started small to have a better understanding of what I'll need